portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,17 +38,27 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Debug
+    // response.getWriter().println(request);
+    // response.getWriter().println(request.getParameter("quantity"));
+
+    
     Query query = new Query("Comment").addSort("postTime", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments = new ArrayList<>();
+    int count = 0;
+    int quantity = Integer.parseInt(request.getParameter("quantity"));
     for (Entity entity : results.asIterable()) {
+      if (count >= quantity) break;
       String message = (String) entity.getProperty("message");
       String username = (String) entity.getProperty("username");
       Date postTime = (Date) entity.getProperty("postTime");
       Comment comment = new Comment(message, username, postTime);
       comments.add(comment);
+      count++;
```

Comment:
Pretty good approach reading records using Iterable then cutting off via counts.  Definitely would be better than loading everything as a List, then truncating in code.

Just curious since I'm not too familiar with Datastore, but does it not require you to "close" the connection if you don't exhaust the iterable?  Other database APIs I've used from before requires something like a `database.close()` since leaving an iterable open would hog up unused resources.

Reply:
`datastore.close()` doesn't work, and I couldn't find anything on closing the datastore connection.